hazardous/_survival_boost.py

```diff
@@ -572,6 +572,137 @@ def predict_survival_function(self, X, times=None):
+    ):
+        """Estimate the conditional incidence functions for each event type
+        knowing that the individual has been censored at a given time :math:`s` i.e.
+        has survived until this time :math:`s`.
+
+        For each incidence function, we return:
+
+        .. math::
+            \forall s > t,
+
+            F_k(t| x_i, T > s) = \mathbb{P}(T \leq t, \Delta = k| X=x_i, T > s)
+            = \frac{\mathbb{P}(s < T \leq t, \Delta = k| X=x_i)}
+            {\mathbb{P}(s < T \leq t | X=x_i)}
+            = \frac{F_k(t| x_i) - F_k(s| x_i)}{S(s| x_i)}
+
+        And:
+            \forall s > t,
+
+            S(t| x_i, T > s) = \mathbb{P}(T \geq t, \Delta = k| X=x_i, T > s)
+            = \frac{\mathbb{P}(T \geq t, \Delta = k| X=x_i)}
+            {\mathbb{P}(s < T \leq t | X=x_i)}
+            = \frac{S(t| x_i)}{S(s| x_i)}
+
+
+        where :math:`F_k(t| x_i)` is the estimated cumulative incidence function for
+        event type :math:`k` at time :math:`t` and :math:`S(s| x_i)` is the estimated
+        survival function at time :math:`s`.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            The feature vectors for each observation for which to estimate the
+            incidence functions knowing the censoring time.
+
+        censored_times : array-like of shape (n_samples,)
+            The time of censoring for each observation.
+
+        times : array-like, default=None
+            The time horizons at which to estimate the probabilities. If ``None``,
+            it uses the grid generated during ``fit`` based on the parameter
+            ``n_time_grid_steps``.
+
+        Returns
+        -------
+        incidence_functions_knowing_censoring_time : ndarray of shape
+            (n_samples, n_events + 1, n_times) The estimated probabilities of the
+            different time horizons. For the times that are lower than the observed
+            event times, the survival function is 1, and the incidence functions will
+            be 0.
+        """
+        if times is None:
+            times = self.time_grid_
+        incidence_functions_knowing_censoring_time = self.predict_cumulative_incidence(
+            X, times=times
+        )
+        predictions_at_censoring_time = [
+            self.predict_cumulative_incidence(
+                X.iloc[i : i + 1], times=np.array([censored_times[i]])
+            )
+            for i in range(len(X))
+        ]
+        predictions_at_censoring_time = np.asarray(
+            predictions_at_censoring_time
+        ).reshape(len(X), len(self.event_ids_), 1)
+
+        # Compute the incidence functions knowing the censoring time.
+        survival_functions = predictions_at_censoring_time[:, 0, :]
+
+        incidence_functions_knowing_censoring_time[
+            :, 1:, :
+        ] -= predictions_at_censoring_time[:, 1:, :]
+        incidence_functions_knowing_censoring_time /= survival_functions[:, :, None]
```

Comment:
I haven't run the code but I assume that's what you want, right?
```suggestion
        incidence_functions_knowing_censoring_time /= survival_functions[:, None, :]
```